unicode.h

```diff
@@ -1,475 +1,1315 @@
+            {0x1d4ad,0x1d4ad},{0x1d4ba,0x1d4ba},{0x1d4bc,0x1d4bc},{0x1d4c4,0x1d4c4},{0x1d506,0x1d506},{0x1d50b,0x1d50c},
+            {0x1d515,0x1d515},{0x1d51d,0x1d51d},{0x1d53a,0x1d53a},{0x1d53f,0x1d53f},{0x1d545,0x1d545},{0x1d547,0x1d549},
+            {0x1d551,0x1d551},{0x1d6a6,0x1d6a7},{0x1d7cc,0x1d7cd},{0x1da8c,0x1da9a},{0x1daa0,0x1daa0},{0x1dab0,0x1deff},
+            {0x1df1f,0x1dfff},{0x1e007,0x1e007},{0x1e019,0x1e01a},{0x1e022,0x1e022},{0x1e025,0x1e025},{0x1e02b,0x1e0ff},
+            {0x1e12d,0x1e12f},{0x1e13e,0x1e13f},{0x1e14a,0x1e14d},{0x1e150,0x1e28f},{0x1e2af,0x1e2bf},{0x1e2fa,0x1e2fe},
+            {0x1e300,0x1e7df},{0x1e7e7,0x1e7e7},{0x1e7ec,0x1e7ec},{0x1e7ef,0x1e7ef},{0x1e7ff,0x1e7ff},{0x1e8c5,0x1e8c6},
+            {0x1e8d7,0x1e8ff},{0x1e94c,0x1e94f},{0x1e95a,0x1e95d},{0x1e960,0x1ec70},{0x1ecb5,0x1ed00},{0x1ed3e,0x1edff},
+            {0x1ee04,0x1ee04},{0x1ee20,0x1ee20},{0x1ee23,0x1ee23},{0x1ee25,0x1ee26},{0x1ee28,0x1ee28},{0x1ee33,0x1ee33},
+            {0x1ee38,0x1ee38},{0x1ee3a,0x1ee3a},{0x1ee3c,0x1ee41},{0x1ee43,0x1ee46},{0x1ee48,0x1ee48},{0x1ee4a,0x1ee4a},
+            {0x1ee4c,0x1ee4c},{0x1ee50,0x1ee50},{0x1ee53,0x1ee53},{0x1ee55,0x1ee56},{0x1ee58,0x1ee58},{0x1ee5a,0x1ee5a},
+            {0x1ee5c,0x1ee5c},{0x1ee5e,0x1ee5e},{0x1ee60,0x1ee60},{0x1ee63,0x1ee63},{0x1ee65,0x1ee66},{0x1ee6b,0x1ee6b},
+            {0x1ee73,0x1ee73},{0x1ee78,0x1ee78},{0x1ee7d,0x1ee7d},{0x1ee7f,0x1ee7f},{0x1ee8a,0x1ee8a},{0x1ee9c,0x1eea0},
+            {0x1eea4,0x1eea4},{0x1eeaa,0x1eeaa},{0x1eebc,0x1eeef},{0x1eef2,0x1efff},{0x1f02c,0x1f02f},{0x1f094,0x1f09f},
+            {0x1f0af,0x1f0b0},{0x1f0c0,0x1f0c0},{0x1f0d0,0x1f0d0},{0x1f0f6,0x1f0ff},{0x1f1ae,0x1f1e5},{0x1f203,0x1f20f},
+            {0x1f23c,0x1f23f},{0x1f249,0x1f24f},{0x1f252,0x1f25f},{0x1f266,0x1f2ff},{0x1f6d8,0x1f6dc},{0x1f6ed,0x1f6ef},
+            {0x1f6fd,0x1f6ff},{0x1f774,0x1f77f},{0x1f7d9,0x1f7df},{0x1f7ec,0x1f7ef},{0x1f7f1,0x1f7ff},{0x1f80c,0x1f80f},
+            {0x1f848,0x1f84f},{0x1f85a,0x1f85f},{0x1f888,0x1f88f},{0x1f8ae,0x1f8af},{0x1f8b2,0x1f8ff},{0x1fa54,0x1fa5f},
+            {0x1fa6e,0x1fa6f},{0x1fa75,0x1fa77},{0x1fa7d,0x1fa7f},{0x1fa87,0x1fa8f},{0x1faad,0x1faaf},{0x1fabb,0x1fabf},
+            {0x1fac6,0x1facf},{0x1fada,0x1fadf},{0x1fae8,0x1faef},{0x1faf7,0x1faff},{0x1fb93,0x1fb93},{0x1fbcb,0x1fbef},
+            {0x1fbfa,0x1ffff},{0x2a6e0,0x2a6ff},{0x2b739,0x2b73f},{0x2b81e,0x2b81f},{0x2cea2,0x2ceaf},{0x2ebe1,0x2f7ff},
+            {0x2fa1e,0x2ffff},{0x3134b,0xe0000},{0xe0002,0xe001f},{0xe0080,0xe00ff},{0xe01f0,0xeffff},{0xffffe,0xfffff},
+            {0x10fffe,0x10ffff}
+    };
+}
+}
+
+namespace REGEX_RANGES {
+    // \s whitespace (81)
+    static const std::vector<std::pair<uint32_t, uint32_t>> Whitespace = {
+            {0x9,0xd},{0x20,0x20},{0x85,0x85},{0xa0,0xa0},{0x1680,0x1680},{0x2000,0x200a},{0x2028,0x2029},
+            {0x202f,0x202f},{0x205f,0x205f},{0x3000,0x3000}
+    };
+}
+
+class UNICODE {

```

Comment:
Can we avoid this `class` and have basic function calls + static containers that are lazy initialized on first use?

Prefix all functions with `unicode_`. For example: `unicode_to_codepoints()

Reply:
This can work, but I'm not sure it's the best way to do it. See my comment below. The good thing about using a class is that we can make many instance of it, each one having different category definitions, and they won't mess with each other.

Reply:
> The good thing about using a class is that we can make many instances of it, each one having different category definitions, and they won't mess with each other.

Ideally, there should be a singe unicode configuration. In what case would we need to have differing category definitions?